lib/pbench/agent/tool_meister.py

```diff
@@ -72,6 +73,85 @@ class ToolException(Exception):
     pass
 
 
+class PermaTool:
+    def __init__(self, name, tool_opts, logger):
+        self.name = name
+        self.tool_opts = tool_opts.split(" ")
+        self.logger = logger
+        self.install_path = None
+        for opt in self.tool_opts:
+            if opt.startswith("--inst="):
+                if opt[len(opt) - 1] == "\n":
+                    self.install_path = opt[7 : len(opt) - 1]
+                else:
+                    self.install_path = opt[7:]
+                self.logger.debug("FOUND")
+            else:
+                self.logger.debug("NOT FOUND SOMEHOW")
+        self.process = None
+        self.failure = False
+
+    def start(self):
+        if self.install_path is None:
+            self.failure = True
+            self.logger.error(
+                "NO INSTALL PATH PROPERLY GIVEN AS PERMATOOL OPTION, see /opt/pbench-agent/nodexporter --help"
+            )
+            return
+
+        if self.name == "node-exporter":
+            self.logger.debug(self.install_path)
+            # args = ["/usr/bin/screen", "-dmS", "node-exp-screen", self.install_path + "/node_exporter"]
+
+            if not os.path.isfile(self.install_path + "/node_exporter"):
+                self.logger.info(
+                    self.install_path + "/node_exporter" + " does not exist"
+                )
+                self.failure = True
+                return 0
+
+            args = [self.install_path + "/node_exporter"]
+            self.process = subprocess.Popen(
+                args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT
+            )
+        elif self.name == "dcgm":
+            os.environ["PYTHONPATH"] = (
+                self.install_path
+                + "/bindings:"
+                + self.install_path
+                + "/bindings/common"
+            )
+
+            if not os.path.isfile(
+                self.install_path + "/samples/scripts/dcgm_prometheus.py"
+            ):
+                self.logger.info(
+                    self.install_path
+                    + "/samples/scripts/dcgm_prometheus.py"
+                    + " does not exist"
+                )
+                self.failure = True
+                return 0
+
+            args = [f"python2 {self.install_path}/samples/scripts/dcgm_prometheus.py"]
```

Comment:
Are we ensuring that `python2` is installed? I don't think we use it anywhere else.

Reply:
As far as I understand from Keshav's work, someone who intends to use DCGM would have to have python2 anyways, and based on what he was given to work with, it appears as though it was necessary. This definitely should not be assumed though, and a check should be performed beforehand, unless someone has a better idea of how to handle it

Reply:
For this commit I would not try to do anything about this.  We have worked hard to remove code where the pbench-agent makes any modifications to the system.  We could add a check to see if `python2` is installed, but if the user has installed dcgm, that package will have already required that.

The problem here comes when that dcgm package changes to start using Python3.

Let's file an issue to track this for a more future proof fix.

Reply:
Tracked in issue #1837.